Gems/Atom/RHI/Code/Include/Atom/RHI/MultiDeviceImagePool.h

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+#pragma once
+
+#include <Atom/RHI.Reflect/ClearValue.h>
+#include <Atom/RHI.Reflect/ImagePoolDescriptor.h>
+#include <Atom/RHI/MultiDeviceImage.h>
+#include <Atom/RHI/MultiDeviceImagePoolBase.h>
+#include <Atom/RHI/ImagePool.h>
+
+namespace AZ
+{
+    namespace RHI
+    {
+        /**
+         * @brief The data structure used to initialize an RHI::MultiDeviceImage on an RHI::MultiDeviceImagePool.
+         */
+        struct MultiDeviceImageInitRequest
+        {
+            MultiDeviceImageInitRequest() = default;
+
+            MultiDeviceImageInitRequest(
+                MultiDeviceImage& image, const ImageDescriptor& descriptor, const ClearValue* optimizedClearValue = nullptr);
+
+            /// The image to initialize.
+            MultiDeviceImage* m_image = nullptr;
+
+            /// The descriptor used to initialize the image.
+            ImageDescriptor m_descriptor;
+
+            /// An optional, optimized clear value for the image. Certain
+            /// platforms may use this value to perform fast clears when this
+            /// clear value is used.
+            const ClearValue* m_optimizedClearValue = nullptr;
+        };
+
+        /**
+         * @brief The data structure used to update contents of an RHI::MultiDeviceImage on an RHI::MultiDeviceImagePool.
+         */
+        struct MultiDeviceImageUpdateRequest
+        {
+            MultiDeviceImageUpdateRequest() = default;
+
+            /// A pointer to an initialized image, whose contents will be updated.
+            MultiDeviceImage* m_image = nullptr;
+
+            /// The image subresource to update.
+            ImageSubresource m_imageSubresource;
+
+            /// The offset in pixels from the start of the sub-resource in the destination image.
+            Origin m_imageSubresourcePixelOffset;
+
+            /// The source data pointer
+            const void* m_sourceData = nullptr;
+
+            /// The source sub-resource layout.
+            MultiDeviceImageSubresourceLayout m_sourceSubresourceLayout;
+        };
+
+        /**
+         * MultiDeviceImagePool is a pool of images that will be bound as attachments to the frame scheduler.
+         * As a result, they are intended to be produced and consumed by the GPU. Persistent Color / Depth Stencil / Image
+         * attachments should be created from this pool. This pool is not designed for intra-frame aliasing.
+         * If transient images are required, they can be created from the frame scheduler itself.
+         */
+        class MultiDeviceImagePool : public MultiDeviceImagePoolBase
```

Comment:
How come MultiDeviceBufferPool inherits from MultiDeviceResourcepool but for images we have MultiDeviceImagePoolBase?

Reply:
We found that interesting as well, but that is the inheritance structure of the single-device classes, `BufferPool` inherits from `ResourcePool` and `ImagePool` inherits from `ImagePoolBase` which inherits from `ResourcePool`.

Reply:
RHI::BufferPool inherits from RHI::BufferPoolBase. So the pattern should match ImagePool. BufferPoolBase has validation code that we may need via MultiDeviceBufferPoolBase